app/lib/frontend/pub_client.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' as convert;
+
+import 'package:http/http.dart';
+
+/// Client interface for accessing some of pub's public APIs.
+class PubClient {
+  final Client _client;
+  final String _siteRoot;
+
+  PubClient(this._client, {String siteRoot = 'https://pub.dartlang.org'})
```

Comment:
If this is a http client, why not just give it a default value? Is there no default global http client?

Reply:
I am not that familiar with the recent changes in the `http` package, and previously, for the `BrowserClient()` to work in the browser, it shouldn't have a codepath to initialize the IO client. I thought I just keep it that way here, keeping the option open for browser-based uses.